### PR TITLE
Process: do not inherit all file descriptors to child processes [try 2]

### PR DESCRIPTION
**Pull request by weissi**

### CODE REVIEW

**Comment by weissi** [comment]
@swift-ci please test

File: TestFoundation/xdgTestHelper/main.swift [hunk 1]
```diff
@@ -264,8 +275,11 @@ case "--nspathfor":
 #if !os(Windows)
 case "--signal-test":
     signalTest()
+
+case "--print-open-file-descriptors":
```
Comment (millenomi) [inline TestFoundation/xdgTestHelper/main.swift]: This breaks the Windows build.

Comment (weissi) [reply TestFoundation/xdgTestHelper/main.swift]: @millenomi are you sure? It's in a `#if !os(Windows)` block

Comment (millenomi) [reply TestFoundation/xdgTestHelper/main.swift]: I missed it right there >.<' I'll double-check the test to make sure the chain holds all the way through.

Comment (millenomi) [reply TestFoundation/xdgTestHelper/main.swift]: It is! I'm not sure if we should #if !os(Windows) the test explicitly, but it works out the way the patch is.

Comment (millenomi) [reply TestFoundation/xdgTestHelper/main.swift]: @weissi can you also guard the test so that it isn't invoked incorrectly?

Comment (weissi) [reply TestFoundation/xdgTestHelper/main.swift]: yes, done. thanks

**Comment by weissi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Comment by millenomi** [comment]
```17:04:18 fatal: destination path 'ninja' already exists and is not an empty directory.```

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Comment by millenomi** [comment]
@shahmishal Nodes are failing independently of the patches with the error above.

**Comment by gmittert** [comment]
@swift-ci please test linux

**Comment by weissi** [comment]
haha corelibs xctest doesn't support `XCTAssertLessThanEqual`

**Comment by weissi** [comment]
@swift-ci please test and merge

**Comment by weissi** [comment]
@swift-ci please test and merge

**Comment by weissi** [comment]
@swift-ci please test and merge

**Comment by weissi** [comment]
compiler crash again

```
macosx-x86_64/XCTest.build/Release/SwiftXCTest.build/Objects-normal/x86_64/WaiterManager.o 
15:55:19 1.	Swift version 5.1.2-dev (Swift dc5f896a92)
15:55:19 2.	While running pass #301731 SILFunctionTransform "SimplifyCFG" on SILFunction "@$s11SwiftXCTest7XCTMainys5NeverOSayAA0B4CaseCm04testE5Class_SaySS_yAFKctG8allTeststGF".
15:55:19  for 'XCTMain(_:)' (at /Users/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-macOS/branch-master/swift-corelibs-xctest/Sources/XCTest/Public/XCTestMain.swift:58:8)
15:55:19 0  swift                    0x000000011158d2c5 llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 37
15:55:19 1  swift                    0x000000011158c238 llvm::sys::RunSignalHandlers() + 248
15:55:19 2  swift                    0x000000011158d8c6 SignalHandler(int) + 278
15:55:19 3  libsystem_platform.dylib 0x00007fff7ca67b5d _sigtramp + 29
15:55:19 4  libsystem_platform.dylib 0x0000000000004e70 _sigtramp + 2203702064
15:55:19 5  libsystem_c.dylib        0x00007fff7c9216a6 abort + 127
15:55:19 6  libsystem_c.dylib        0x00007fff7c8ea20d basename_r + 0
15:55:19 7  swift                    0x00000001117c1f71 (anonymous namespace)::CastEmitter::emit((anonymous namespace)::Source, (anonymous namespace)::Target) (.cold.12) + 33
15:55:19 8  swift                    0x000000010ddc09fb (anonymous namespace)::CastEmitter::emit((anonymous namespace)::Source, (anonymous namespace)::Target) + 4187
15:55:19 9  swift                    0x000000010ddbf98a (anonymous namespace)::CastEmitter::emitAndInjectIntoOptionals((anonymous namespace)::Source, (anonymous namespace)::Target, unsigned int) + 186
15:55:19 10 swift                    0x000000010ddbe464 swift::emitSuccessfulScalarUnconditionalCast(swift::SILBuilder&, swift::ModuleDecl*, swift::SILLocation, swift::SILValue, swift::SILType, swift::CanType, swift::CanType, swift::SILInstruction*) + 388
15:55:19 11 swift                    0x000000010ddbe2d0 swift::emitSuccessfulScalarUnconditionalCast(swift::SILBuilder&, swift::SILLocation, swift::SILDynamicCastInst) + 128
15:55:19 12 swift                    0x000000010dce954d swift::CastOptimizer::simplifyCheckedCastBranchInst(swift::CheckedCastBranchInst*) + 1037
15:55:19 13 swift                    0x000000010dc77d1c (anonymous namespace)::SimplifyCFG::simplifyBlocks() + 4204
15:55:19 14 swift                    0x000000010dc7347b (anonymous namespace)::SimplifyCFG::run() + 7051
15:55:19 15 swift                    0x000000010dc718b1 (anonymous namespace)::SimplifyCFGPass::run() + 881
15:55:19 16 swift                    0x000000010dba420f swift::SILPassManager::runPassOnFunction(unsigned int, swift::SILFunction*) + 1711
15:55:19 17 swift                    0x000000010dba4e6f swift::SILPassManager::runFunctionPasses(unsigned int, unsigned int) + 1055
15:55:19 18 swift                    0x000000010dba5f0e swift::SILPassManager::execute() + 606
15:55:19 19 swift                    0x000000010d73927b swift::SILPassManager::executePassPipelinePlan(swift::SILPassPipelinePlan const&) + 187
15:55:19 20 swift                    0x000000010dbad5b5 swift::runSILOptimizationPasses(swift::SILModule&) + 229
15:55:19 21 swift                    0x000000010d7fe34d swift::CompilerInstance::performSILProcessing(swift::SILModule*, swift::UnifiedStatsReporter*) + 653
15:55:19 22 swift                    0x000000010d5ea802 performCompileStepsPostSILGen(swift::CompilerInstance&, swift::CompilerInvocation&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule> >, bool, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::PrimarySpecificPaths const&, bool, int&, swift::FrontendObserver*, swift::UnifiedStatsReporter*) + 1378
15:55:19 23 swift                    0x000000010d5e0700 performCompile(swift::CompilerInstance&, swift::CompilerInvocation&, llvm::ArrayRef<char const*>, int&, swift::FrontendObserver*, swift::UnifiedStatsReporter*) + 9328
15:55:19 24 swift                    0x000000010d5dd316 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 3046
15:55:19 25 swift                    0x000000010d57ab0e main + 734
15:55:19 26 libdyld.dylib            0x00007fff7c87c3d5 start + 1
15:55:19 error: Abort trap: 6 (in target 'SwiftXCTest' from project 'XCTest')
15:55:19 
15:55:19 
<unknown>:0: error: unable to execute command: Abort trap: 6
15:55:19 <unknown>:0: error: compile command failed due to signal 6 (use -v to see invocation)
```

**Comment by weissi** [comment]
@swift-ci please test and merge

**Comment by weissi** [comment]
@swift-ci please test and merge

**Comment by weissi** [comment]
@millenomi I always get a compiler crash here whilst compiling XCTest. seen this before?

**Comment by weissi** [comment]
@swift-ci please test and merge

**Comment by compnerd** [comment]
This broke Foundation on android
https://dev.azure.com/compnerd/windows-swift/_build/results?buildId=18592&view=logs&j=1d59321a-8fbc-5405-c03e-4696bd0a260e&t=1075a828-3280-5930-19a8-eb2ba31fb39b&l=783

**Comment by weissi** [comment]
@compnerd oh, no getdtablesize on Android... I’ll make a followup fixing this for Android later today.

**Comment by spevans** [comment]
@weissi  You should be able to use `/proc/self/fd` to get the list of open filedescriptors on linux, so may work on android as well

**Comment by weissi** [comment]
@compnerd https://github.com/apple/swift-corelibs-foundation/pull/2586

**Pull request merged**